### PR TITLE
Connect the profile service to the profile storage.  Inject initial state into `BrowserWindow` instances.

### PR DESCRIPTION
**Pull request by ncalexan**
r? @rnewman, @bgrins

### CODE REVIEW

File: app/main/actions/profile-actions.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+import * as profileActionTypes from './../constants/profile-action-types';
+
+/**
+ * The set of bookmarks is different.
+ */
+export function bookmarkSet(bookmarks) {
+  return {
+    type: profileActionTypes.DID_SET_BOOKMARKED_SET,
```
Comment (rnewman) [inline app/main/actions/profile-actions.js]: Anything but `SET`? `DID_CHANGE_BOOKMARKS`?

File: app/main/browser.js [hunk 2]
```diff
@@ -183,14 +195,13 @@ ipc.on('instrument-event', (event, args) => {
   instrument.event(args.name, args.method, args.label, args.value);
 });
 
-ipc.on('new-window', () => createWindow());
+ipc.on('new-window', async function() {
+  await createWindow();
```
Comment (rnewman) [inline app/main/browser.js]: Consider commenting why this isn't just `, createWindow);`

**Comment by rnewman** [comment]
LGTM.

File: app/main/browser.js [hunk 3]
```diff
@@ -57,25 +62,23 @@ function sendToAllWindows(event, args) {
...
+    // TODO: maintain this list in storage.
     const recentBookmarks = currentState.bookmarks
       .valueSeq()
       .sortBy(bookmark => bookmark.createdAt)
       .take(5);
     BrowserMenu.build({ bookmarks: recentBookmarks });
   }
-  if (force || !previousState || currentState.locations !== previousState.locations) {
+
+  if (!previousState || currentState.locations !== previousState.locations) {
```
Comment (bgrins) [inline app/main/browser.js]: This should probably be using immutable equality methods (same for bookmarks)

File: app/main/reducers/location-reducers.js [hunk 4]
```diff
@@ -10,21 +10,16 @@
...
+import * as profileActionTypes from '../constants/profile-action-types';
 
 import Immutable from 'immutable';
 
-export default function locationReducer(state, command) {
-  if (typeof state === 'undefined') {
-    return Immutable.Map();
-  }
+export default function locationReducer(state = new Immutable.Map(), action) {
+  const payload = action.payload;
```
Comment (bgrins) [inline app/main/reducers/location-reducers.js]: This'll need to be rebased on top of the new, better fake results at: https://github.com/mozilla/tofino/commit/8bd95b5a8da91eff7cf66f2807f73d9d6dd84189#diff-2918c2484fc2347d14dfdfdcb2d8bf1bL24

**Comment by ncalexan** [comment]
Landed!

**Pull request closed without merging**